src/idl_parser.cpp

```diff
@@ -2679,20 +2686,6 @@ CheckedError Parser::ParseDecl(const char *filename) {
                          flatbuffers::numeric_limits<voffset_t>::max());
       for (voffset_t i = 0; i < static_cast<voffset_t>(fields.size()); i++) {
         auto &field = *fields[i];
-        const auto &id_str = field.attributes.Lookup("id")->constant;
```

Comment:
And If in this way was accepted, Attribute [documentations ](https://google.github.io/flatbuffers/md__schemas.html) would be updated as well.

Reply:
This is something we cannot change. 

I think for proto->fbs generation, we always export the file id attribute for all fields. That part should be straightforward.

About the consecutive invariant, this is a bit more tricky. We need to handle a few things IMO:

- Throw an error when a gap in proto field ids is detected
  - This needs to handle proto `reserved` ids. I'm not sure if we can just skip those field id value, probably need to test this to see it works. 
  - Alternatively, we can generate a dummy field for each reserved id (name it like `_reserved{#}:bool (deprecated, id: {#}`) 
  - This is not 100% accurate, as reserved fields can be reused (https://developers.google.com/protocol-buffers/docs/proto3#reserved) but this is typically not how they are used.
  -

Reply:
@dbaileychess 

- missing or set twice`  issue is straightforward, you are right. I got confused because [test.proto](https://github.com/google/flatbuffers/blob/ae6662374d9dd1dd297ab83f2946cafd4af8572f/tests/prototest/test.proto#L42) has same ids and it is not correct from protobuf respective.
- protobuf allow to use ids that are not consecutive. Generally, can we ignore consecutive ids ? Just sort them and map to 0 .. n.
- I will work on reserved id.

And another issue not really related to this p.r., proto3 does not have default value. It supported in proto2. I think we should consider it.

Reply:
Puede integrar complementos no?

Reply:
@dbaileychess I need your advice. I have faced with some problems here:

1. In proto to fbs, When we have a map in proto file, we create an equivalent Table with key and value field. This key and value do not have id, so it has conflict with `either all fields or no fields must have an 'id' attribute` statement.
2. In proto to fbs, for oneof (union field) field, in proto does not have id. same problem in section 1.
3.  And finally, about reserved ids, Reserved ids can not be reused, so I think we can just throw exception if reserved id is used.

My solution for these  problems is like this:
Solve problem in fbs -> code (cpp, java,...). I mean ignore consecutive, invariant and either all fields or no fields must have an 'id' attribute. We just sort ids, and assign them 0 to N

Reply:
will get back to this, still discussing internally what the best course of action is.

Reply:
So I think the best course of action is to do what you suggest, sort by ids and assign them to 0 to N. 

We should also provide a similar thing to our `.fbs` schema evolution with the `--conform` flag. We should check if two protos conform to each other and throw an error if they don't. For example, if there was a proto with a gap of IDs and later they update it by adding field with an ID within that gap, that would lead to a conform failure. We would still allow the generation of the schema to occur, but we have at least warned the user that it isn't compatible.

Reply:
@dbaileychess There are 2 approaches, First one, we sort and map the ids  in parser, and second sort and map in fbs generator.
The first approach which is currently implemented in this PR, will allow all fbs files have non-consecutive and non-invariant, and missing ids.
The second approach will affect just protobuf -> fbs process.

I prefer second approach, I mean fix ids problem in fbs-generator.